bastion/src/supervisor.rs

```diff
@@ -700,6 +778,19 @@ impl Supervisor {
                     killed,
                     bcast.id()
                 );
+
+                match restart_strategy_inner {
+                    ActorRestartStrategy::ExponentialBackOff {
+                        timeout,
+                        multiplier,
+                    } => {
+                        let start_in =
+                            timeout.as_secs() + (timeout.as_secs() * multiplier * restart_count);
```

Comment:
Perhapse it make a sense to mention about it in the docs as well, what do you think?

Reply:
I think it would! :)

Reply:
Is it possible to add linear backoff too. If we can do that inside this PR that would be nice.

Reply:
I could add the `LinearBackOff` struct into the `ActorRestartStrategy` enum with the similar signature (that will contain only the timeout). Also, I'd like to say that the if we pass the `multiplayer: 0` and any `timeout` value for the exponential back off, the supervisor will try to restart the failed actor at regular intervals. It's just a corner use case for this type of exponential back off :)

Reply:
Another thing that I though about is to add the `limit`/`attempts` option to the `ExponentialBackOff` struct that has the `Option<u32>` type. It gives a way to stop recovering failed actor if we actually can't do.
However, I'm not sure, is supported this feature right now (to stop recovering after N attempts)

Reply:
yes, in other frameworks and langs it is working like that. 
https://github.com/trendmicro/backoff-python#backoffon_exception
That would be nice to have max_retries. Which enables us to resolve #105 :)

Reply:
> I could add the LinearBackOff struct into the ActorRestartStrategy enum with the similar signature (that will contain only the timeout).

Yes please, that will be clear what is the difference. It is a corner case but let's be explicit rather than implicit. :)

Reply:
@vertexclique I'm thinking about the adding the max_retries in code. Does it make a sense to append the `max_restarts` for the `Supervisor`/`Children` types instead of re-ogranizing it in the struct? (like in the code below)
```
pub struct RestartStrategy {
    max_restarts: Option<usize>
    strategy: ActorRestartStrategy
}
```

Reply:
@Relrin Actually what you are suggesting is better since they are logically separate. Naming also looks good. If you want to do that in this PR just ping us. We can close two issues at the same time. Or we can implement this in other PR. This is your call…